src/main/java/ludogorie_soft/reservations_platform_api/service/impl/CalendarServiceImpl.java

```diff
@@ -0,0 +1,184 @@
+package ludogorie_soft.reservations_platform_api.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import ludogorie_soft.reservations_platform_api.entity.Booking;
+import ludogorie_soft.reservations_platform_api.entity.Property;
+import ludogorie_soft.reservations_platform_api.exception.APIException;
+import ludogorie_soft.reservations_platform_api.repository.BookingRepository;
+import ludogorie_soft.reservations_platform_api.repository.PropertyRepository;
+import ludogorie_soft.reservations_platform_api.service.CalendarService;
+import net.fortuna.ical4j.data.CalendarBuilder;
+import net.fortuna.ical4j.data.CalendarOutputter;
+import net.fortuna.ical4j.data.ParserException;
+import net.fortuna.ical4j.model.Calendar;
+import net.fortuna.ical4j.model.Component;
+import net.fortuna.ical4j.model.component.CalendarComponent;
+import net.fortuna.ical4j.model.component.VEvent;
+import net.fortuna.ical4j.model.property.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.net.URL;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CalendarServiceImpl implements CalendarService {
+
+    private final PropertyRepository propertyRepository;
+    private final BookingRepository bookingRepository;
+
+    private static final String BASE_URL = "http://localhost:8080/calendar/ical/";
+
+    @Value("${booking.ics.airBnb.directory}")
+    private String icsAirBnbDirectory;
+
+    @Value("${booking.ics.myCal.directory}")
+    private String icsMyCalDirectory;
+
+    public String getMyCalendar(Long propertyId) throws IOException {
```

Comment:
Can we think of a better method name here? Seems to me that this method is like `generateCalendarFile()`

Reply:
Done!